src/modules/typst.rs

```diff
@@ -0,0 +1,104 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::typst::TypstConfig;
+use crate::formatter::{StringFormatter, VersionFormatter};
+
+/// Creates a module with the current Typst version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("typst");
+    let config = TypstConfig::try_load(module.config);
+
+    let is_hs_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_hs_project {
```

Comment:
I think this name fits the context better:
```suggestion
    let is_typst_project = context
        .try_begin_scan()?
        .set_files(&config.detect_files)
        .set_extensions(&config.detect_extensions)
        .set_folders(&config.detect_folders)
        .is_match();

    if !is_typst_project {
```